lib/bundler/source/rubygems.rb

```diff
@@ -84,10 +86,10 @@ def install(spec)
 
         # Download the gem to get the spec, because some specs that are returned
         # by rubygems.org are broken and wrong.
-        if spec.source_uri
+        if spec.remote
           # Check for this spec from other sources
-          uris = [spec.source_uri.without_credentials]
-          uris += source_uris_for_spec(spec)
+          uris = [spec.remote.anonymized_uri]
+          uris += remotes_for_spec(spec).map { |remote| remote.anonymized_uri }
```

Comment:
I think I'll turn this into an array of remotes instead of an array of URIs in a future change. Need to implement `==` for `Remote` first.